go/libraries/doltcore/diff/diff_summary.go

```diff
@@ -0,0 +1,338 @@
+// Copyright 2019 Liquidata, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package diff
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+
+	humanize "github.com/dustin/go-humanize"
+
+	"github.com/liquidata-inc/dolt/go/cmd/dolt/cli"
+	"github.com/liquidata-inc/dolt/go/cmd/dolt/errhand"
+	"github.com/liquidata-inc/dolt/go/store/atomicerr"
+	"github.com/liquidata-inc/dolt/go/store/datas"
+	"github.com/liquidata-inc/dolt/go/store/diff"
+	"github.com/liquidata-inc/dolt/go/store/types"
+	"github.com/liquidata-inc/dolt/go/store/util/status"
+)
+
+type diffFunc func(changeChan chan<- types.ValueChanged, stopChan <-chan struct{})
+
+// Summary prints a summary of the diff between two values
+func Summary(ctx context.Context, v1, v2 types.Value, colLen int) errhand.VerboseError {
+	if is1, err := datas.IsCommit(v1); err != nil {
+		return errhand.BuildDError("").AddCause(err).Build()
+	} else if is1 {
+		if is2, err := datas.IsCommit(v2); err != nil {
+			return errhand.BuildDError("").AddCause(err).Build()
+		} else if is2 {
+			cli.Println("Comparing commit values")
+
+			var err error
+			v1, _, err = v1.(types.Struct).MaybeGet(datas.ValueField)
+			if err != nil {
+				return errhand.BuildDError("").AddCause(err).Build()
+			}
+
+			v2, _, err = v2.(types.Struct).MaybeGet(datas.ValueField)
+			if err != nil {
+				return errhand.BuildDError("").AddCause(err).Build()
+			}
+		}
+	}
+
+	var singular, plural string
+	if v1.Kind() == v2.Kind() {
+		switch v1.Kind() {
+		case types.StructKind:
+			singular = "Field"
+			plural = "Fields"
+		case types.MapKind:
+			singular = "Entry"
+			plural = "Entries"
+		default:
+			singular = "Value"
+			plural = "Values"
+		}
+	}
```

Comment:
Is this part necessary if `Summary` is only being used for table diffs and the `rowData` getting passed in is a `Map`? It's copied from noms

Reply:
if you change your function signature to use types.Map for v1 and v2 then you ensure the calling code only passes maps.  Then you can kill a lot of code such as diffSummaryList, diffSummarySet, etc.

Reply:
Fixed and removed all those functions/code having to do with types